### PR TITLE
feat(stoneintg-842): Link plr console URL into gitlab commit status

### PR DESCRIPTION
**Pull request by 14rcole**
This change adds a link to the Konflux console in the pipeline view on gitlab _if_ the cluster has a console (determined by the value of CONSOLE_URL in the integration service manager pod.

A sample pipeline can be viewed [here](https://gitlab.cee.redhat.com/rycole/webhook-test/-/pipelines/8040272). Note: I hardcoded the URL for the pod since there was no console in the cluster being used to test the change.

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: controllers/statusreport/statusreport_adapter.go [hunk 1]
```diff
@@ -72,7 +72,10 @@ func NewAdapter(context context.Context, snapshot *applicationapiv1alpha1.Snapsh
 // which (indirectly) triggered its execution.
 func (a *Adapter) EnsureSnapshotTestStatusReportedToGitProvider() (controller.OperationResult, error) {
 	if gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot) {
+		a.logger.Info("Snapshot is created by PAC push event, skipping 'EnsureSnapshotTestStatusReportedToGitProvider'")
 		return controller.ContinueProcessing()
+	} else {
+		a.logger.Info("Snapshot is not created by PAC push event, running 'EnsureSnapshotTestStatusReportedToGitProvider'")
```
Comment (MartinBasti) [inline controllers/statusreport/statusreport_adapter.go:78]: There is log.info later showing which reporter will be used, do we really need this log message?

**Review by MartinBasti — COMMENTED** [review]
Should we have an unittest?

File: controllers/statusreport/statusreport_adapter.go (hunk 1 above)
Comment (14rcole) [reply controllers/statusreport/statusreport_adapter.go]: This was a debugging statement I forgot to remove.  Fixed.

**Comment by 14rcole** [comment]
@dirgim @MartinBasti I brought this up during a standup a while back.  The code can't be unit tested as-is.  All of  the changed functions are private and the `_test.go` files are in a different package (`status_test` instead of `status`).  During our last discussion we agreed that this testing issue should be addressed but that its outside the scope of this ticket.  If you no longer feel that way I can either
1. Make the functions being changed here public and add unit tests
2. Change the package name in the test files to give them access to private functions.  This would be a big change so I wouldn't recommend this as part of STONEINTG-842

**Comment by MartinBasti** [comment]
```diff
diff --git a/status/reporter_gitlab_test.go b/status/reporter_gitlab_test.go
index 8c1115a..2f2d21a 100644
--- a/status/reporter_gitlab_test.go
+++ b/status/reporter_gitlab_test.go
@@ -211,6 +211,27 @@ var _ = Describe("GitLabReporter", func() {
                                })).To(Succeed())
                })
 
+               It("creates a commit status for snapshot with TargetURL in CommitStatus", func() {
+
+                       PipelineRunName := "TestPipeline"
+                       expectedURL := status.FormatPipelineURL(PipelineRunName, hasSnapshot.Namespace, logr.Discard())
+
+                       muxCommitStatusPost(mux, sourceProjectID, digest, expectedURL)
+                       muxMergeNotes(mux, sourceProjectID, mergeRequest, "")
+                       muxCommitStatusesGet(mux, sourceProjectID, digest, nil)
+
+                       Expect(reporter.ReportStatus(
+                               context.TODO(),
+                               status.TestReport{
+                                       FullName:            "fullname/scenario1",
+                                       ScenarioName:        "scenario1",
+                                       TestPipelineRunName: PipelineRunName,
+                                       Status:              integrationteststatus.IntegrationTestStatusInProgress,
+                                       Summary:             "summary",
+                                       Text:                "detailed text here",
+                               })).To(Succeed())
+               })
+
                It("does not create a commit status or comment for snapshot with existing matching checkRun in running state", func() {
                        summary := "Integration test for snapshot snapshot-sample and scenario scenario1 is running"

```

**Review by MartinBasti — APPROVED** [review]

**Review by dirgim — APPROVED** [review]
LGTM

**Review by jsztuka — APPROVED** [review]

**Comment by dirgim** [comment]
@14rcole the code looks all set, but needs a rebase on latest main to pass EC checks.

**Pull request merged**